types/src/traits.rs

```diff
@@ -110,7 +110,7 @@ macro_rules! array_impls {
 }
 
 array_impls! {
-	1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16
+	0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16
```

Comment:
This is kind of a tangent, but we could just use const generics now that they are available instead of the macro:

```rust
impl<P: Serialize, const N: usize> ToRpcParams for [P; N] {}
```

Reply:
The problem I think is that serde impls for `[T; 0] .... [T; 32]` and `impl<P: Serialize, const N: usize> ToRpcParams for [P; N]` for 0...usize::MAX?

Reply:
ok, maybe we should open a PR to serde then?!

Reply:
```rust
impl<P, const N: usize> ToRpcParams for [P; N] where [P; N]: Serialize {}
```
Works for me